src/BizHawk.Emulation.Cores/Consoles/Nintendo/SNES/LibsnesCore.cs

```diff
@@ -22,7 +22,8 @@ namespace BizHawk.Emulation.Cores.Nintendo.SNES
 	public unsafe partial class LibsnesCore : IEmulator, IVideoProvider, ISaveRam, IStatable, IInputPollable, IRegionable, ICodeDataLogger,
 		IDebuggable, ISettable<LibsnesCore.SnesSettings, LibsnesCore.SnesSyncSettings>, IBSNESForGfxDebugger
 	{
-		[CoreConstructor(VSystemID.Raw.SGB)]

```

Comment:
...and here.

Reply:
At the very least this one is questionable as the core doesn't ever actually sets its `SystemID` to SGB. I don't really mind this too much, but I do think we should be able to agree on what this attribute is supposed to mean.

Reply:
Semantically, it indicates which constructor to use for a given `RomGame`.
Also, the set across all of the core's constructors doubles as the list of sysIDs which that core can emulate. We could split that out to the `[Core]` attr instead, but I think a single-source-of-truth is better.
https://github.com/TASEmulators/BizHawk/blob/dbc6acc8fb6b98ad3ce3b9a415f6a5566471eb66/src/BizHawk.Emulation.Cores/Consoles/Nintendo/SNES/LibsnesCore.cs#L112-L115
^ This is clearly incorrect. I imagine it was a hack to get the frontend to show the right menus/tools.

Reply:
> the set across all of the core's constructors doubles as the list of sysIDs which that core can emulate.

This is precisely the issue; I don't think this invariant necessarily needs to hold true (it doesn't even right now with that old bsnes core), and even with your suggested change bsnes* core cannot provide the GB/C SystemID, even though they have a corresponding CoreConstructor attribute.

That attribute is currently entirely internal and only used for the internal selection process of which cores to consider for a given rom, so putting semantic stuff here that is supposed to communicate meaning to the reader while doing nothing for the core or romloader that's using it is misguided imho.
It's fine to communicate this SGB stuff with a comment, but the attribute itself doesn't need to be here.

The conceptual question here is whether it is acceptable for a core to load a rom of type `X` and declare a SystemID of type `Y != X` using that loaded rom, and I say yes.

Reply:
How about this:
- `RomLoader` never sets `SGB` on any rom (since any GB rom could be used with SGB, they should all be detected as the same, and `GB`/`GBC` is the better choice)
- There is no core preference for `SGB`, only `GB` and maybe `GBC`
- Cores are perfectly honest with setting `IEmulator.SystemId` (not necessarily matching the one passed in, since a sync setting can choose between `GB`/`GBC`/`SGB`)
- `SystemId` always matches a `[CoreConstructor]`—if it doesn't make sense to have that ctor, like with `SGB` which `RomLoader` will never use, it can simply be `private` and call something like `this(default)`

I think that would give the consistency I'm after while being simpler for the end-user.

Reply:
The first three should already be present here.

If you feel like adding dummy constructors with the SGB attribute just to signal intent, fine.

Reply:
The first 3 are not already implemented. This PR addresses some of them, but I'm saying we should clean it up on master first.